api/src/domain/tax-filings/taxFilingsInterfaceFactory.ts

```diff
@@ -23,36 +23,54 @@ export const taxFilingsInterfaceFactory = (apiTaxFilingClient: TaxFilingClient):
   };
 
   const onboarding = async (request: taxFilingInterfaceRequest): Promise<UserData> => {
-    const userData = await lookup(request);
-    if (userData.taxFilingData.state === "UNREGISTERED") {
-      const response = await apiTaxFilingClient.onboarding(
-        request.taxId,
-        userData.user.email,
-        request.businessName
-      );
-      const registered = ["SUCCESS", "PENDING"].includes(response.state);
-      return {
-        ...userData,
-        taxFilingData: {
-          ...userData.taxFilingData,
-          businessName: request.businessName,
-          lastUpdatedISO: new Date(Date.now()).toISOString(),
-          registeredISO: registered ? new Date(Date.now()).toISOString() : undefined,
-          errorField: response.errorField,
-          state: response.state == "SUCCESS" ? "PENDING" : response.state,
-        },
-      };
-    } else {
-      const registered = ["SUCCESS", "PENDING"].includes(userData.taxFilingData.state ?? "");
-      return {
-        ...userData,
-        taxFilingData: {
-          ...userData.taxFilingData,
-          registeredISO:
-            userData.taxFilingData.registeredISO ??
-            (registered ? new Date(Date.now()).toISOString() : undefined),
-        },
-      };
+    const response = await apiTaxFilingClient.onboarding(
+      request.taxId,
+      request.userData.user.email,
+      request.businessName
+    );
```

Comment:
Not something that you need to do for this story, but this function taking 3 string types as a parameter list is a code smell. If any of them wind up out of order, the whole thing won't work and we wouldn't even get a type failure. This is a great candidate for a Named Parameters refactoring